src/main/java/com/iggroup/universityworkshopmw/domain/services/OpenPositionsService.java

```diff
@@ -0,0 +1,88 @@
+package com.iggroup.universityworkshopmw.domain.services;
+
+import com.iggroup.universityworkshopmw.domain.exceptions.NoAvailableDataException;
+import com.iggroup.universityworkshopmw.domain.model.OpenPosition;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+@Slf4j
+@Component
+public class OpenPositionsService {
+   private Map<String, List<OpenPosition>> clientPositionStore;
+
+   public OpenPositionsService() {
+      clientPositionStore = new HashMap<>();
+   }
+
+   public List<OpenPosition> getOpenPositionsForClient(String clientId) throws NoAvailableDataException {
+      List<OpenPosition> openPositions = clientPositionStore.get(clientId);
+
+      if (openPositions != null) {
+         return openPositions;
+      }
+      throw new NoAvailableDataException("No open positions exist for client: " + clientId);
+   }
+
+   public void addOpenPositionForClient(String clientId, OpenPosition openPosition) {
+      List<OpenPosition> openPositions = clientPositionStore.get(clientId);
+      if (openPositions != null) {
+         openPositions.add(openPosition);
+      } else {
+         clientPositionStore.put(clientId, newArrayList(openPosition));
+      }
+   }
+
+   public Double closeOpenPosition(String clientId, String openPosition, Double closingPrice) throws NoAvailableDataException {
+      List<OpenPosition> openPositions = clientPositionStore.get(clientId);
+      if (clientId == null) {
+         throw new NoAvailableDataException("Client had no positions");
+      }
+
+      OpenPosition position = openPositions.stream()
+         .filter(pos -> pos.getId().equals(openPosition))
+         .findFirst()
+         .orElseThrow(() -> new NoAvailableDataException("No position exists with id: " + openPosition));
+
+      Double closingProfitAndLoss = getNewProfitAndLoss(closingPrice, position.getOpeningPrice(), position.getBuySize());
+      openPositions.remove(position);
+      if (openPositions.isEmpty()) {
+         clientPositionStore.remove(clientId);
+      }
+      return closingProfitAndLoss;
```

Comment:
when we calculate the new profit and loss after closing, we need to update client funds